localstack/services/awslambda/resource_providers/aws_lambda_alias.py

```diff
@@ -0,0 +1,172 @@
+# LocalStack Resource Provider Scaffolding v2
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Optional, Type, TypedDict
+
+import localstack.services.cloudformation.provider_utils as util
+from localstack.services.cloudformation.resource_provider import (
+    CloudFormationResourceProviderPlugin,
+    OperationStatus,
+    ProgressEvent,
+    ResourceProvider,
+    ResourceRequest,
+)
+
+
+class LambdaAliasProperties(TypedDict):
+    FunctionName: Optional[str]
+    FunctionVersion: Optional[str]
+    Name: Optional[str]
+    Description: Optional[str]
+    Id: Optional[str]
+    ProvisionedConcurrencyConfig: Optional[ProvisionedConcurrencyConfiguration]
+    RoutingConfig: Optional[AliasRoutingConfiguration]
+
+
+class ProvisionedConcurrencyConfiguration(TypedDict):
+    ProvisionedConcurrentExecutions: Optional[int]
+
+
+class VersionWeight(TypedDict):
+    FunctionVersion: Optional[str]
+    FunctionWeight: Optional[float]
+
+
+class AliasRoutingConfiguration(TypedDict):
+    AdditionalVersionWeights: Optional[list[VersionWeight]]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class LambdaAliasProvider(ResourceProvider[LambdaAliasProperties]):
+
+    TYPE = "AWS::Lambda::Alias"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[LambdaAliasProperties],
+    ) -> ProgressEvent[LambdaAliasProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/Id
+
+        Required properties:
+          - FunctionName
+          - FunctionVersion
+          - Name
+
+        Create-only properties:
+          - /properties/Name
+          - /properties/FunctionName
+
+        Read-only properties:
+          - /properties/Id
+
+
+
+        """
+        model = request.desired_state
+        lambda_ = request.aws_client_factory.lambda_
+
+        create_params = util.select_attributes(
+            model, ["FunctionName", "FunctionVersion", "Name", "Description", "RoutingConfig"]
+        )
+
+        ctx = request.custom_context
+        if not ctx.get(REPEATED_INVOCATION):
+            result = lambda_.create_alias(**create_params)
+            model["Id"] = result["AliasArn"]
+            ctx[REPEATED_INVOCATION] = True
+
+            if model.get("ProvisionedConcurrencyConfig"):
+                lambda_.put_provisioned_concurrency_config(
+                    FunctionName=model["FunctionName"],
+                    Qualifier=model["Id"].split(":")[-1],
+                    ProvisionedConcurrentExecutions=model["ProvisionedConcurrencyConfig"][
+                        "ProvisionedConcurrentExecutions"
+                    ],
+                )
+
+            return ProgressEvent(
+                status=OperationStatus.IN_PROGRESS,
+                resource_model=model,
+            )
```

Comment:
We'll need to check if the context on AWS is something you can mutate because otherwise this provider wouldn't work on AWS since you're not passing the context back with the progress event. :thinking: 

For now please also pass the context object on return.